src/Stack/Options.hs

```diff
@@ -279,6 +280,12 @@ configOptsParser hide0 =
            <> help "Extra directories to check for libraries"
            <> hide
             )))
+    <*> optionalFirst (textOption
+             ( long "with-ghc"
```

Comment:
In `Stack.Types.Config` you define `configMonoidOverrideGccPathName = "with-gcc"`, so you could potentially reuse that definition instead of duplicating the literal.